velox/functions/sparksql/aggregates/tests/ModeAggregateTest.cpp

```diff
@@ -0,0 +1,502 @@
+
+    createDuckDbTable({vectors});
+    testAggregations({vectors}, {}, {"mode(c0)"}, "SELECT mode(c0) FROM tmp");
+
+    testAggregations({vectors}, {}, {"mode(c1)"}, "SELECT mode(c1) FROM tmp");
+  }
+
+  void testMode(
+      const std::string& expression,
+      const std::vector<std::string>& groupKeys,
+      const VectorPtr& vector1,
+      const VectorPtr& vector2,
+      const RowVectorPtr& expected) {
+    auto vectors = makeRowVector({vector1, vector2});
+    testAggregations({vectors}, groupKeys, {expression}, {expected});
+  }
+};
+
+TEST_F(ModeAggregateTest, groupByInteger) {
+  vector_size_t num = 37;
+
+  auto vector1 = makeFlatVector<int32_t>(
+      num, [](vector_size_t row) { return row % 3; }, nullEvery(4));
+  auto vector2 = makeFlatVector<int32_t>(
+      num, [](vector_size_t row) { return row % 2; }, nullEvery(5));
+
+  testModeWithDuck(vector1, vector2);
+
+  // Test when some group-by keys have only null values.
+  auto vector3 =
+      makeNullableFlatVector<int64_t>({1, 1, 1, 2, 2, 2, 3, 3, std::nullopt});
+  auto vector4 = makeNullableFlatVector<int64_t>(
+      {10, 10, 10, 20, std::nullopt, 20, std::nullopt, std::nullopt, 20});
+
+  testModeWithDuck(vector3, vector4);
+}
+
+TEST_F(ModeAggregateTest, groupByDouble) {
+  vector_size_t num = 37;
+
+  auto vector1 = makeFlatVector<int32_t>(
+      num, [](vector_size_t row) { return row % 3; }, nullEvery(4));
+  auto vector2 = makeFlatVector<double>(
+      num, [](vector_size_t row) { return row % 2 + 0.05; }, nullEvery(5));
+
+  testModeWithDuck(vector1, vector2);
+}
+
+TEST_F(ModeAggregateTest, groupByBoolean) {
+  vector_size_t num = 37;
+
+  auto vector1 = makeFlatVector<int32_t>(
+      num, [](vector_size_t row) { return row % 3; }, nullEvery(4));
+  auto vector2 = makeFlatVector<bool>(
+      num, [](vector_size_t row) { return row % 2 == 0; }, nullEvery(5));
+
+  auto expected = makeRowVector(
+      {makeNullableFlatVector<int32_t>({std::nullopt, 0, 1, 2}),
+       makeFlatVector<bool>({true, false, false, false})});
+
+  testMode("mode(c1)", {"c0"}, vector1, vector2, expected);
+}
+
+TEST_F(ModeAggregateTest, groupByTimestamp) {
+  vector_size_t num = 10;
+
+  auto vector1 = makeFlatVector<int32_t>(
+      num, [](vector_size_t row) { return row % 3; }, nullEvery(4));
+  auto vector2 = makeFlatVector<Timestamp>(
+      num,
+      [](vector_size_t row) { return Timestamp{row % 2, 17'123'456}; },
+      nullEvery(5));
+
+  auto expected = makeRowVector(
+      {makeNullableFlatVector<int32_t>({std::nullopt, 0, 1, 2}),
+       makeFlatVector<Timestamp>(
+           {Timestamp{0, 17'123'456},
+            Timestamp{1, 17'123'456},
+            Timestamp{1, 17'123'456},
+            Timestamp{0, 17'123'456}})});
+
+  testMode("mode(c1)", {"c0"}, vector1, vector2, expected);
+}
+
+TEST_F(ModeAggregateTest, groupByDate) {
+  vector_size_t num = 10;
+
+  auto vector1 = makeFlatVector<int32_t>(
```

Comment:
It might be more readable to write out the values:

```
makeFlatVector<int32_t>({0, 1, 2, 0, 1, 2, 0, 1, 2, 0}};
```

> vector1

Avoid naming variables a1, a2, a3.

Reply:
Updated, thanks.